apps/camera/js/app.js

```diff
@@ -261,16 +264,30 @@ App.prototype.onCriticalPathDone = function() {
   this.emit('criticalpathdone');
 };
 
+/**
+ * Lazy load activity dependencies.
+ *
+ * @private
+ */
+App.prototype.onActivity = function() {
+  if (!this.criticalPathDone) {
+    this.controllers.lazy.push('controllers/confirm');
+  } else {
+    this.loadLazyControllers(['controllers/confirm']);
```

Comment:
As far as I can tell, this code path is never reached, not with a reused camera process, nor with a fresh one. What seems to happen is the existing camera process will restart itself entirely if already opened when a pick from MMS is required.